lib/api.js

```diff
@@ -145,11 +145,15 @@ class JSDOM {
   }
 
   static fromFile(filename, options = {}) {
-    return Promise.resolve().then(() => {
+    return new Promise((resolve, reject) => {
```

Comment:
I think this will break cause an error to be thrown, instead of a promise to be rejected, if an invalid filename is given. (I.e. one that causes URL parsing failures.) I'm surprised this wasn't tested.

A better solution here would be to move to async functions.

Reply:
Actually, a thrown error inside a `Promise` constructor’s `executor` callback will still result in a rejected `Promise`: https://tc39.es/ecma262/#sec-promise-executor

This only has bearing on the execution timing (`executor` is executed synchronously, whereas `then` callbacks are executed as part of the job queue).